### PR TITLE
Require repo scope for PATs for private repos and basic authentication

### PR DESCRIPTION
**Pull request by jolheiser**
> The scoped token PR just checked all API routes but in fact, some web routes like `LFS`, git `HTTP`, container, and attachments supports basic auth. This PR added scoped token check for them.

### CODE REVIEW

**Review by techknowlogick — APPROVED** [review]

File: modules/context/permission.go [hunk 1]
```diff
@@ -106,3 +110,32 @@ func RequireRepoReaderOr(unitTypes ...unit.Type) func(ctx *Context) {
...
 }
+
+// RequireRepoScopedToken check whether personal access token has repo scope
+func CheckRepoScopedToken(ctx *Context, repo *repo_model.Repository) {
+	if !ctx.IsBasicAuth || ctx.Data["IsApiToken"] != true {
+		return
+	}
+
+	var err error
+	scope, ok := ctx.Data["ApiTokenScope"].(auth_model.AccessTokenScope)
```
Comment (wxiaoguang) [inline modules/context/permission.go:121]: If no test covers it, it is very fragile.

Maybe next refactoring changes the key "ApiTokenScope", then this "check" will become a noop, the accesses just pass it.

**Review by delvh — APPROVED** [review]

**Comment by wxiaoguang** [comment]
This PR breaks my LFS client.

Reverting these 2 "lfs/*.go" files , then  my client works again.

<details>

gitea-app_1     | 2023/04/27 20:37:14 ...rvices/lfs/server.go:547:authenticate() [W] [644a6c7a-2] Authentication failure for provided token with Error: invalid token claim
gitea-app_1     | 2023/04/27 20:37:14 [644a6c7a-2] router: completed GET /org/repo.git/info/lfs/objects/xxxxxxxxxxxxxxxxxxxxxxxxxx for 1.2.3.4:0, 401 Unauthorized in 5.9ms @ lfs/server.go:80(lfs.DownloadHandler)

</details>

File: services/lfs/server.go [hunk 2]
```diff
@@ -86,6 +86,11 @@ func DownloadHandler(ctx *context.Context) {
 		return
 	}
 
+	repository := getAuthenticatedRepository(ctx, rc, true)
```
Comment (wxiaoguang) [inline services/lfs/server.go:89]: DownloadHandler: getAuthenticatedRepository(requireWrite=true) ?

Comment (lunny) [reply services/lfs/server.go:89]: This lines should be removed.

**Pull request merged**